realbigsean/mock-relay
# Add Deneb support

## Test dependencies:
- `lighthouse`: [this branch](https://github.com/jimmygchen/lighthouse/pull/11), contains patched `mev-rs` dependencies, and scripts updated for testing builder flow

## Steps to test:
1. Check out the above lighthouse branch and build using `make && make install-lcli`
2. Update `vars.env`
    - `BN_COUNT` to 5 and `VC_COUNT` to 4 so we run an extra client pair (`node-5`) for the mock relay
    - Change `SECONDS_PER_SLOT` and `SECONDS_PER_ETH1_BLOCK` to `12` (`mock_relay` defaults to `mainnet`)
3. Clone and build geth devnet-8 fork: https://github.com/lightclient/go-ethereum/tree/devnet-8
4. Start local testnet
5. Run mock relay (the below connects to beacon node 5)

```
cargo run --bin mock-relay -- \
  --beacon-node http://localhost:8005 \
  --execution-endpoint http://localhost:5005 \
  --genesis-fork-version 0x42424242 \
  --jwt-secret $HOME/.lighthouse/local-testnet/geth_datadir5/geth/jwtsecret
```

## Additional Information

- Notes for testing: it seems like builder flow doesn't get triggered until epoch 4, so there will be a bit of initial wait.

## TODO

Once this setup is working:
- [x] Create `mev-rs` PR for [this branch](https://github.com/jimmygchen/mev-rs/tree/deserialize_deneb_block)
  - https://github.com/ralexstokes/mev-rs/pull/124
- [x] Create `lighthouse` PR for [this branch](https://github.com/jimmygchen/lighthouse/pull/11)
  - https://github.com/sigp/lighthouse/pull/4710
